src/objects/viewport.ts

```diff
@@ -138,11 +143,14 @@ async function startViewportDateRange(): Promise<void> {
 
 async function updateViewportDateRange(): Promise<void> {
   const viewport = await waitForViewport()
-  const { scaleX } = await waitForScales()
-  const left = scaleX.invert(viewport.left)
-  const right = scaleX.invert(viewport.right)
+  const scales = await waitForScales()
+  const left = scales.getPositionFromXPixels(viewport.left)
+  const right = scales.getPositionFromXPixels(viewport.right)
+
+  if (left instanceof Date && right instanceof Date) {
```

Comment:
Starting to see the different modes show up in types. In `dag` mode the x scale is not date based. So checking here if the left/right values are dates before emitting them up.